docs/operations/hook-sidecar.md

```diff
@@ -0,0 +1,168 @@
+# Hook Sidecar Container
+
+## Introduction
+
+In KubeVirt, a Hook Sidecar container is a sidecar container (a secondary container that runs along with the main
+application container within the same Pod) used to apply customizations before the Virtual Machine is initialized. This
+ability is provided since 100% of the [libvirt domain XML](https://libvirt.org/formatdomain.html) elements and
+attributes are not configurable/customizable using the VMI specification.
+
+The sidecar containers communicate with the main container over a socket with a gRPC protocol. There are two main
+sidecar hooks:
+
+1. `onDefineDomain`: This hook helps to customize libvirt's XML and return the new XML over gRPC for the VM creation.
+2. `preCloudInitIso`: This hook helps to customize the cloud-init configuration. It operates on and returns JSON
+   formatted cloud-init data.
+
+## Enabling Sidecar feature gate
+
+To be able to use the sidecar hooks, the Sidecar feature gate must be enabled on the cluster. If you're using a
+development cluster backed by KubeVirt CI, you could enable it using the following steps:
+```shell
+# export FEATURE_GATES=<feature-gate-1>,<feature-gate-2>
+# e.g. to enable Sidecar and HotplugNICs feature gates run below
+$ export FEATURE_GATES=Sidecar,HotplugNICs
+$ make cluster-sync
+```
+
+For non-development clusters, it can be enabled by following the steps mentioned in [Activating feature
+gates](../activating_feature_gates).
+
+## Sidecar-shim container image
+
+To reduce the amount of boilerplate that developers need to do in order to run VM with custom modifications, we provide
+the [sidecar-shim-image](https://quay.io/repository/kubevirt/sidecar-shim) that takes care of implementing the
+communication with the main container.
+
+The image contains the `sidecar-shim` binary built using
+[`sidecar_shim.go`](https://github.com/kubevirt/kubevirt/blob/main/cmd/sidecars/sidecar_shim.go) which should be kept
+as the entrypoint of the container. This binary will search in `$PATH` for binaries named after the hook names (e.g
+`onDefineDomain` and `preCloudInitIso`) and run them. User must provide the necessary arguments as command line options
+(flags).
+
+In the case of `onDefineDomain`, the arguments will be the VMI information as JSON string, (e.g `--vmi vmiJSON`) and
+the current domain XML (e.g `--domain domainXML`). As standard output it expects the modified domain XML.
```

Comment:
Yes, it sounds the same to me. Do you suggest changing it what you asked? Why? They already sound the same thing to me.

NOTE: I'm not a native English speaker. :)

Reply:
From how I read it, "As a standard output" is an adverbial phrase. It is the broader category in which "XML" is one of several options. It adds additional context to the sentence, and, as such, having it start the sentence makes it a bit awkward to read because the reader has to remember the context until it reaches the part of the sentence to which it applies. 

I'd say the best way to write this is: "It expects modified domain XML as standard output." (The subject > verb > object > context) But what the "It" is is not entirely clear. Is it the binary?

Reply:
Thanks for the explanation.

> I'd say the best way to write this is: "It expects modified domain XML as standard output." (The subject > verb > object > context) But what the "It" is is not entirely clear. Is it the binary?

The "it" is the 'onDefineDomain' hook. We are starting the paragraph with `In the case of onDefineDomain`. So, to me, it seems that the "it" part is clear. Let me know if you think being explicit would be better.